Chained.h

```diff
@@ -11,197 +13,195 @@
                     ++numWith1;
                 }
                 else{
                     ++numWithMore;
                 }
             }
-            for(auto kvp : (*hashmap)[i]){
-                if(print) {
-                    printKeyValuePair(kvp);
-                }
-            }
-            if(print) {
-                std::cout << "\n";
-            }
         }
         std::cout << std::endl;
 
-        //stats output
         if(elements.empty()){
             std::cout << "No Elements\n";
         } else {
             std::sort(elements.begin(), elements.end());
-            Statistic* elementStats = evaluateStats(&elements);
-            std::cout << "Total Elements: " << elementStats->total << "\n";
-            std::cout << "Mean Elements: " << elementStats->mean << "\n";
-            std::cout << "Stdev Elements: " << elementStats->stdev << "\n";
+            Statistic elementStats = evaluateStats(&elements);
+            std::cout << "Total Elements: " << elementStats.total << "\n";
+            std::cout << "Mean Elements: " << elementStats.mean << "\n";
+            std::cout << "Stdev Elements: " << elementStats.stdev << "\n";
 
             std::sort(collisions.begin(), collisions.end());
-            Statistic* collisionStats = evaluateStats(&collisions);
-            std::cout << "Total Collisions: " << collisionStats->total << "\n";
-            std::cout << "Mean Collisions: " << collisionStats->mean << "\n";
-            std::cout << "Stdev Collisions: " << collisionStats->stdev << "\n";
+            Statistic collisionStats = evaluateStats(&collisions);
+            std::cout << "Total Collisions: " << collisionStats.total << "\n";
+            std::cout << "Mean Collisions: " << collisionStats.mean << "\n";
+            std::cout << "Stdev Collisions: " << collisionStats.stdev << "\n";
 
-            std::cout << "Total Number of Lists: " << hashmapSize << "\n";
+            std::cout << "Total Number of Lists: " << hashmap_.size() << "\n";
             std::cout << "Number of Lists With Zero Elements: " << numWith0 << "\n";
             std::cout << "Number of Lists with One Element: " << numWith1 << "\n";
             std::cout << "Number of Lists with more than One Element: " << numWithMore << "\n";
-            //I could make a display of all element sizes. Box plot? Scatterplot?
+            //Histogram?
 
             std::cout << "Number of elements in the biggest List: " << elements[elements.size()-1] << "\n";
             std::cout << "Number of elements in the smallest List: " << elements[0] << "\n";
         }
+        std::cout << std::endl;
     }
-};
 
-template<>
-uint64_t ChainedMap<std::string, uint64_t>::defaultGetReturn() {
-    return -1;
-}
-template<>
-Product* ChainedMap<uint64_t, Product*>::defaultGetReturn(){
-    return nullptr;
-}
+    double getLoadFactor() override {
+        return (static_cast<float>(getNumPairs()))/(static_cast<float>(hashmap_.size()));
```

Comment:
Once you explicitly cast the first number and the return type is float/double, then everything else will get type "promoted".  But the cast is safe.